### PR TITLE
Added extra configuration anotation options and localized the config gui

### PR DESCRIPTION
**Pull request by Runesmacher**
Fancyfied the config pages.
Made sure that potion array can't be less then 256, maybe top limit might have to be bigger.
Hide the version config in the config gui

image

### CODE REVIEW

**Comment by Runesmacher** [comment]
Once this is merged, i can create a pull request for Integrated Dynamics, which requires these changes to be present

**Comment by rubensworks** [comment]
It looks like something went wrong with your spacing. In some places you use tabs and on some places you use tabs. The convention in this project is to always use 4 spaces.

The localization keys you propose should be scoped with mod id, just like the blocks and items, because otherwise they might conflict with other mods. There also shouldn't be spaces in such localization keys.

File: src/main/resources/assets/cyclopscore/lang/en_US.lang [hunk 1]
```diff
@@ -31,3 +31,47 @@ multiblock.cyclopscore.error.hollow.air=Position %s must be hollow.
...
+config.cyclopscore.general.analytics=Send anonymous analytics
+config.cyclopscore.general.anonymousAnalyticsID=Anonymous id
+config.cyclopscore.general.crashOnInvalidRecipe=Crash on invalid Recipe
+config.cyclopscore.general.crashOnModCompatCrash=Crash on Compatibility issue
+config.cyclopscore.general.debug=Develop mode
+config.cyclopscore.general.minimumPotionTypesArraySize=Minimal size of potion array
+config.cyclopscore.general.version=Version
+config.cyclopscore.general.versionChecker=Enable version check
+
+## Mod compatibiliy settings
```
Comment (rubensworks) [inline src/main/resources/assets/cyclopscore/lang/en_US.lang]: Typo: "compatibility"

File: src/main/java/org/cyclops/cyclopscore/config/ConfigProperty.java [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import org.cyclops.cyclopscore.config.configurable.IConfigurable;
 import org.cyclops.cyclopscore.config.extendedconfig.ExtendedConfig;
 import org.cyclops.cyclopscore.init.ModBase;
+import org.lwjgl.Sys;
```
Comment (rubensworks) [inline src/main/java/org/cyclops/cyclopscore/config/ConfigProperty.java]: Why is this here? I think it can be removed.

File: src/main/java/org/cyclops/cyclopscore/client/gui/config/GuiConfigOverviewBase.java [hunk 3]
```diff
@@ -28,7 +29,7 @@
 	 */
 	public GuiConfigOverviewBase(ModBase mod, GuiScreen parentScreen) {
 		super(parentScreen, getConfigElements(mod), mod.getModId(), false, false,
-			GuiConfig.getAbridgedConfigPath(mod.getConfigHandler().getConfig().toString()));
+		        mod.getModName() + " " + L10NHelpers.localize("config.cyclopscore"));
```
Comment (rubensworks) [inline src/main/java/org/cyclops/cyclopscore/client/gui/config/GuiConfigOverviewBase.java]: I would add the mod name as a parameter to the localization, because in some languages the mod name could be placed _after_ "configuration". (using %s in the l10n value).

File: src/main/resources/assets/cyclopscore/lang/en_US.lang [hunk 4]
```diff
@@ -31,3 +31,47 @@ multiblock.cyclopscore.error.hollow.air=Position %s must be hollow.
 
 # Vanilla additions
 tile.air.name=Air
+
+# Config Screens
+config.cyclopscore=configuration
+
+## Config categories
+config.cyclopscore.item=Item settings
+config.cyclopscore.item.tooltip=Change settings of items
```
Comment (rubensworks) [inline src/main/resources/assets/cyclopscore/lang/en_US.lang]: Currently these tooltips don't really give any useful additional information. If there's nothing special about a category, I would remove the tooltip.

**Review by rubensworks — CHANGES_REQUESTED** [review]
Added some minor comments.
Could you look into squashing the commits so that there is one commit that adds the annotation options, and one that adds the l10n?

**Comment by rubensworks** [comment]
Closed in favor of #30
